src/blockdata/witness.rs

```diff
@@ -27,23 +28,35 @@ pub struct Witness {
     /// Number of elements in the witness.
     /// It is stored separately (instead of as VarInt in the initial part of content) so that method
     /// like [`Witness::push`] doesn't have case requiring to shift the entire array
-    witness_elements: u64,
+    witness_elements: usize,
```

Comment:
we may want to save 8 bytes in the Witness struct  using u32 for `witness_elements`, `last` and `second_to_last` and still support MAX_VECTOR_SIZE witnesses, however I thought using usize was a safer bet initially

Reply:
`last` and `second_to_last` are not reset when calling the clear function